examples/blend/main.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2023 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	"image/color"
+	_ "image/png"
+	"log"
+	"math"
+
+	"golang.org/x/image/font/inconsolata"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/examples/resources/images/blend"
+	"github.com/hajimehoshi/ebiten/v2/text"
+)
+
+const (
+	screenWidth  = 780
+	screenHeight = 600
+)
+
+// mode is a blend mode with description.
+type mode struct {
+	blend ebiten.Blend
+	name  string
+}
+
+// Game is a canvas for drawing blend mode tiles.
+type Game struct {
+	source      *ebiten.Image
+	dest        *ebiten.Image
+	offscreen   *ebiten.Image
+	tileSize    float64
+	gridW       float64
+	gridH       float64
+	tileSpacing float64 // Horizontal and vertical margin between tiles.
+	textSpacing float64 // Additional vertical margin for text.
```

Comment:
Use ints instead of float64s for all the members. (float64 is necessary only when passing them to GeoM)

Reply:
In general, we should avoid float values in logics as float values can include numerical errors.

Reply:
After all, they will still have to be cast in float64 for a smooth resize. Am I wrong?

Reply:
Oh you are right. At least `tileSpacing` and `textSpacing` can be float64. I still think the others should be integers.

I realized `tileSpacing` and `textSpacing` can be consts (= untyped numeric values) instead of Game's members. What do you think?

Reply:
Yeah, actually we can bring every setting to the package level and keep only resources and calculated values in the Game.

Reply:
> Oh you are right. At least tileSpacing and textSpacing can be float64. I still think the others should be integers.

Idk it makes too many casts from scratch.

`float64(g.gridW)*(float64(g.tileSize)+g.tileSpacing)-g.tileSpacing, float64(g.gridH)*(float64(g.tileSize)+g.tileSpacing)`

And we can't keep everything int too cause it will cause jitter.

Reply:
Why not

```go
gw, gh := g.gridW*(g.tileSize+g.tileSpacing)-g.tileSpacing, g.gridH*(g.tileSize+g.tileSpacing)
```

Reply:
> Yeah, actually we can bring every setting to the package level

You don't have to define the consts in the package level. Defining them in Draw is fine.

Reply:
> Why not
> 
> ```go
> gw, gh := g.gridW*(g.tileSize+g.tileSpacing)-g.tileSpacing, g.gridH*(g.tileSize+g.tileSpacing)
> ```

Because you said, that g.gridW, g.gridH and g.tileSize should be integers.
If we keep them all float64 it will be the same as you wrote (but removing floats from util function makes sense).

Reply:
Why do gw and gh have to be float64?

Reply:
Hmm, they don't have division, right...
I need to think about it a bit.

Reply:
Fixed. I removed most floats I had.